src/nvim/eval/funcs.c

```diff
@@ -940,6 +940,55 @@ static void f_char2nr(typval_T *argvars, typval_T *rettv, FunPtr fptr)
       (const char_u *)tv_get_string(&argvars[0]));
 }
 
+// "charidx()" function
+static void f_charidx(typval_T *argvars, typval_T *rettv, FunPtr fptr)
+{
+  const char *str;
```

Comment:
Vim does this to support older compilers but local variable scoping is preferred.

https://neovim.io/develop/style-guide.xml#Local_Variables